### PR TITLE
feat(platform): Add instructions field to agent submissions

### PR DESCRIPTION
**Pull request by majdyz**
## Summary

Added an optional "Instructions" field for agent submissions to help users understand how to run agents and what to expect.

image

image

image

### Changes Made

**Backend:**
- Added `instructions` field to `AgentGraph` and `StoreListingVersion` database models
- Updated `StoreSubmission`, `LibraryAgent`, and related Pydantic models
- Modified store submission API routes to handle instructions parameter
- Updated all database functions to properly save/retrieve instructions field
- Added graceful handling for cases where database doesn't yet have the field

**Frontend:**
- Added instructions field to agent submission flow (PublishAgentModal)
- Positioned below "Recommended Schedule" section as specified
- Added instructions display in library/run flow (RunAgentModal)  
- Positioned above credentials section with informative blue styling
- Added proper form validation with 2000 character limit
- Updated all TypeScript types and API client interfaces

### Key Features

- ✅ Optional field - fully backward compatible
- ✅ Proper positioning in both submission and run flows
- ✅ Character limit validation (2000 chars)
- ✅ User-friendly display with "How to use this agent" styling
- ✅ Only shows when instructions are provided

### Testing

- Verified Pydantic model validation works correctly
- Confirmed schema validation enforces character limits
- Tested graceful handling of missing database fields
- Code formatting and linting completed

## Test plan

- [ ] Test agent submission with instructions field
- [ ] Test agent submission without instructions (backward compatibility)
- [ ] Verify instructions display correctly in run modal
- [ ] Test character limit validation
- [ ] Verify database migrations work properly

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

**Comment by AutoGPT-Agent** [comment]
Thanks for this well-structured PR implementing the instructions field for agent submissions! The changes look comprehensive, covering both backend and frontend components with proper validation and backward compatibility.

### What looks good:
- Clear description of changes with appropriate references to the Linear ticket
- Comprehensive implementation across backend and frontend
- Proper validation (2000 character limit)
- Backward compatibility handling
- Nice UI implementation with informative styling

### Before this can be merged:
- Please complete the test plan checklist by checking off each item after you've verified it

The implementation itself looks solid, with appropriate schema updates, form fields, and UI components. Once you've completed the test plan verification and checked off those items, this should be ready for merging.

**Comment by AutoGPT-Agent** [comment]
Thank you for submitting this PR to add the instructions field to agent submissions. The implementation looks solid, with good coverage across backend, frontend, and database layers.

The changes appear well-designed:
- Adding an optional field maintains backward compatibility
- The UI placement follows the requirements
- Character limits are properly enforced
- The field is handled appropriately in all relevant components

Before this can be approved, please complete the test plan checklist in your PR description. Your test plan items are good, but they need to be checked off to indicate you've actually performed the testing. Once you've verified each test case, please mark the checkboxes as completed.

Additionally, consider adding a brief note about what database changes this introduces - while I can see the schema.prisma changes, a note about whether any migration is needed would be helpful.

Overall, this is a well-structured PR that adds useful functionality to help users understand how to run agents.

**Comment by AutoGPT-Agent** [comment]
Thank you for this well-structured PR that adds an instructions field to agent submissions. The changes look comprehensive, touching both backend and frontend components to implement this new feature.

The code changes themselves look good - you've properly:
- Added the field to all relevant database models
- Updated the Pydantic models and API interfaces
- Added proper validation with character limits
- Added a clean UI component to display instructions in the run modal
- Included database migrations
- Added graceful handling for backward compatibility

However, before we can approve this PR for merging, could you please **check off all the items in your test plan checklist**? Our process requires all checkboxes to be completed to confirm you've performed the necessary testing.

Your test plan includes these items that need to be checked off:
- Test agent submission with instructions field
- Test agent submission without instructions (backward compatibility)
- Verify instructions display correctly in run modal
- Test character limit validation
- Verify database migrations work properly

Once you've completed these tests and checked all the boxes, we'll be able to move forward with the PR.

**Comment by AutoGPT-Agent** [comment]
Thanks for this well-structured PR adding the instructions field to agent submissions! The implementation looks comprehensive, covering both backend and frontend changes consistently.

A couple of things to address before we can merge this:

1. **Checklist needs completion**: Please check off all the items in your test plan. Our process requires all checkboxes to be marked as completed before merging.

2. **Character limit validation**: I see you've added a 2000 character limit with proper validation on the frontend, which is good. Just make sure the backend validation also enforces this same limit to maintain consistency.

Otherwise, this looks like a solid implementation:
- The database migration is well-structured
- You've handled the backward compatibility case properly
- The UI implementation with the blue informative styling looks appropriate
- The field is properly positioned as optional in both the submission and run flows

Please complete the checklist and let us know when it's ready for another review!

**Comment by AutoGPT-Agent** [comment]
Thanks for your detailed PR! The implementation for adding the instructions field to agent submissions looks good and is well-documented.

A few points of feedback:

1. Your PR description is informative and clearly explains the changes, but please use our standard PR template checklist format with proper checkboxes (`- [ ]`) for the test plan section. While your test plan content is good, it should follow our standard format.

2. I noticed you've added a debug output in the frontend: `Debug: instructions = {JSON.stringify(agent.instructions)}` - is this intended to remain in production code? Consider removing this before merging.

3. For the frontend implementation, the `AgentRunsView.tsx` file has both a "Setup Instructions" section that appears to be for debugging and a properly styled "How to use this agent" section in the modal. Make sure only the intended UI elements remain.

Otherwise, this is a well-structured implementation of the instructions field across the entire stack - database models, migrations, API changes, and frontend components all look consistent and follow our patterns. The validation with 2000 character limit and optional field handling is a nice touch.

**Comment by AutoGPT-Agent** [comment]
Thank you for submitting this PR to add an instructions field to agent submissions! The implementation looks comprehensive, covering both backend and frontend aspects of the feature.

## Issues to Fix

1. **Unchecked Test Plan Items**: Your test plan in the PR description has unchecked items. Please execute the test plan and check off the boxes to confirm all tests have been completed.

2. **Debug Code**: There appears to be debug elements in the frontend code that should be removed:
   - A fixed position red debug box at the top of the screen in AgentRunsView.tsx
   - A red "Setup Instructions DEBUG" section showing raw instructions data

## Suggestions

- Consider adding validation tests for the new instructions field to ensure it correctly handles edge cases (empty string, very long text, etc.)
- The 2000 character limit mentioned in your description is implemented in the frontend validation, which is good. Make sure backend validation also enforces this limit.

Once you've completed the test plan and removed the debug elements, this PR should be ready for approval. The core implementation for adding the instructions field looks solid and well-executed.

**Comment by AutoGPT-Agent** [comment]
Thanks for this well-structured PR adding the instructions field to agent submissions! The implementation looks comprehensive, covering both frontend and backend components.

A couple of items to address before merging:

1. Please use the standard PR template checklist format and check off the items to confirm you've completed them. The test plan you've included is good but needs to be part of the properly formatted checklist.

2. Could you verify that the test plan items have been completed? If so, please mark them as checked in the updated PR description.

The implementation itself looks solid:
- You've added the field to all necessary database models
- The migration script looks correct
- Frontend implementation shows the instructions in appropriate places
- Validation with character limits is implemented
- Backward compatibility is maintained

Once you update the PR description with the proper checklist format and mark the completed items, this should be ready to merge.

**Comment by AutoGPT-Agent** [comment]
Thank you for this comprehensive PR adding an instructions field to agent submissions! The implementation looks well thought out across both frontend and backend components.

### What looks good:
- Clear PR description with screenshots
- Well-organized changes across backend models, database migrations, and frontend components
- Good UI implementation showing instructions in both submission and run flows
- The instructions field is optional and backward compatible
- Form validation with character limits

### One issue to address before merging:
- The test plan in your PR description has checkboxes that aren't checked off. Our process requires all checkboxes to be checked before merging. Please go through your test plan, verify each item, and check off the boxes to confirm you've tested each scenario.

Once you've completed and checked off the test plan items, this PR should be ready for merging. The implementation itself looks solid!

**Comment by AutoGPT-Agent** [comment]
Thank you for adding the instructions field to agent submissions. This will definitely help users understand how to run agents and what to expect.

Your implementation looks comprehensive, covering both backend and frontend changes. I appreciate the detailed PR description with screenshots and the clear explanation of changes made.

However, before this PR can be approved:

- Please check off all items in your test plan checklist. You've listed the test points but haven't marked them as completed. According to our PR process, we need all checklist items to be fully checked off.

A few technical observations:

1. The instructions field implementation looks good with proper validation (2000 char limit) and optional nature for backward compatibility.

2. Nice job with the UI implementation - the positioning of the instructions field in the publish modal and the styled display in the run modal both look well-designed.

3. The database migration appears correctly structured to add the new field to both AgentGraph and StoreListingVersion tables.

4. I noticed you've handled the case where the database might not have the field yet with `getattr(sub, "instructions", None)`, which is a good defensive approach.

Once you've marked your test checklist as completed, this PR should be ready for approval.

**Comment by AutoGPT-Agent** [comment]
Thank you for your work on adding the instructions field to agent submissions! This is a valuable feature that will help users better understand how to use agents.

### Feedback

**Positive aspects:**
- Your implementation is comprehensive, covering both backend and frontend changes
- The migrations and database model updates look well-structured
- The UI implementation with appropriate styling looks good
- The feature is backward compatible as the field is optional

**Issues to address:**

1. **Missing checklist items**: Your PR description is missing the complete checklist from the template. Please add the complete checklist and check off the items that you've completed. This should include:
   - Code change checklist
   - Test plan with checked items
   - Configuration changes checklist (if applicable)

2. **Merge conflicts**: The PR has conflicts that need to be resolved before it can be merged.

Once you've updated the PR description with the complete checklist and resolved the merge conflicts, this PR should be ready for another review. The implementation itself looks solid.

**Comment by AutoGPT-Agent** [comment]
Thank you for this well-structured PR adding an instructions field to agent submissions! The changes look comprehensive, covering both frontend and backend implementation with proper UI placement and database schema updates.

The code implementation looks good, with:
- Proper database schema updates and migrations
- Frontend components to display instructions in appropriate places
- Character limits and validation
- Backward compatibility handling

However, before we can merge this PR, please complete the test plan checklist in your PR description. You've created a good test plan with specific items to test, but the checkboxes aren't marked as completed. Please verify you've tested each item and check them off.

Once the test plan checklist is completed, this PR should be ready for merge.

File: autogpt_platform/frontend/src/app/(platform)/library/agents/[id]/components/AgentRunsView/components/RunAgentModal/components/ModalHeader/ModalHeader.tsx [hunk 1]
```diff
@@ -26,6 +29,42 @@ export function ModalHeader({ agent }: ModalHeaderProps) {
...
+                agent.recommended_schedule_cron,
+              ).toLowerCase()}
+            </p>
+          </div>
+        )}
+
+        {/* Setup Instructions */}
+        {agent.instructions && (
+          <div className="mt-4 flex items-start gap-2">
+            <svg
```
Comment (Abhi1992002) [inline autogpt_platform/frontend/src/app/(platform)/library/agents/[id]/components/AgentRunsView/components/RunAgentModal/components/ModalHeader/ModalHeader.tsx]: It would be great if we used this icon from [phosphor](https://phosphoricons.com/?q=info) to ensure consistency in our icons.

**Review by Abhi1992002 — COMMENTED** [review]
Everything looks good to me, but we need to make a minor change. They’re not very important, so I’ve approved it.

File: autogpt_platform/frontend/src/app/(platform)/library/agents/[id]/components/AgentRunsView/components/RunAgentModal/components/ModalHeader/ModalHeader.tsx (hunk 1 above)
Comment (majdyz) [reply autogpt_platform/frontend/src/app/(platform)/library/agents/[id]/components/AgentRunsView/components/RunAgentModal/components/ModalHeader/ModalHeader.tsx]: Good point, fixed

**Review by Abhi1992002 — APPROVED** [review]
